packages/m2m-event-manager/src/services/m2mEventReaderServiceSQL.ts

```diff
@@ -101,6 +112,12 @@ export function m2mEventReaderServiceSQLBuilder(
                 eq(agreementInM2MEvent.producerId, requester),
                 eq(agreementInM2MEvent.producerDelegateId, requester)
               ),
+            }),
+            delegationIdFilter(agreementInM2MEvent, delegationId, {
+              nullFilter: or(
+                eq(agreementInM2MEvent.producerId, requester),
+                eq(agreementInM2MEvent.consumerId, requester)
+              ),
```

Comment:
There could be agreements which I'm the producer delegate and the direct consumer. Should those be included? 
```
        - null: Return only events for resources accessible without delegation
            (public resources, resources accessible as consumer/producer, owned resources, etc.)
```
I think yes, those agreements should be included, since they can be treated as owned resources. However, I would change the openAPI delegationId description because the fact that we are returning events without delegation, in these rare cases, could be technically inaccurate and could lead to confusion